NICE.Identity.Management/ClientApp/src/views/OrganisationsList/OrganisationsList.tsx

```diff
@@ -0,0 +1,451 @@
+					? querystringObject.page[0]
+					: querystringObject.page
+				: 1,
+		);
+		let itemsPerPage = querystringObject.amount
+			? Array.isArray(querystringObject.amount)
+				? querystringObject.amount[0]
+				: querystringObject.amount
+			: 25;
+		const sortedBy = querystringObject.sort
+			? Array.isArray(querystringObject.sort)
+				? querystringObject.sort[0]
+				: querystringObject.sort
+			: "alpha-asc";
+
+		itemsPerPage = Number(itemsPerPage) ? Number(itemsPerPage) : itemsPerPage;
+
+		this.state = {
+			path: querystring,
+			organisations: [],
+			isLoading: true,
+			pageNumber: pageNumber,
+			itemsPerPage: itemsPerPage,
+			sortedBy: sortedBy,
+		};
+
+		document.title = "NICE Accounts - Organisations list";
+	}
+
+	async componentDidMount(): Promise<void> {
+		this.setState({ isLoading: true });
+
+		let organisations = await fetchData(Endpoints.organisationsList);
+
+		if (isDataError(organisations)) {
+			this.setState({ error: organisations });
+		} else {
+			organisations = this.sortOrganisations(
+				this.state.sortedBy,
+				organisations,
+			);
+		}
+
+		this.setState({
+			organisations,
+			isLoading: false,
+		});
+	}
+
+	pastPageRange = (
+		itemsPerPage: string | number,
+		pageNumber: number,
+		dataCount: number,
+	): number => {
+		let pastPageRange = false;
+
+		if (Number(itemsPerPage)) {
+			itemsPerPage = Number(itemsPerPage);
+			pastPageRange = pageNumber * itemsPerPage >= dataCount;
+		}
+
+		if (pastPageRange || itemsPerPage === "all") {
+			pageNumber = 1;
+		}
+
+		return pageNumber;
+	};
+
+	filterOrganisationsBySearch = async (searchQuery: string): Promise<void> => {
+		this.setState({ isLoading: true });
+
+		let organisations = await fetchData(
+			`${Endpoints.organisationsList}?q=${searchQuery}`,
+		);
+
+		if (isDataError(organisations)) {
+			this.setState({ error: organisations });
+		}
+
+		organisations = this.sortOrganisations(this.state.sortedBy, organisations);
+
+		const itemsPerPage = Number(this.state.itemsPerPage)
+			? Number(this.state.itemsPerPage)
+			: this.state.itemsPerPage;
+
+		const pageNumber = this.pastPageRange(
+			itemsPerPage,
+			this.state.pageNumber,
+			this.state.organisations.length,
+		);
+
+		this.setState({
+			organisations,
+			searchQuery,
+			pageNumber,
+			isLoading: false,
+		});
+	};
+
+	getPaginateStartAndFinishPosition = (
+		websitesCount: number,
+		pageNumber: number,
+		itemsPerPage: number | string,
+	): { start: number; finish: number } => {
+		const paginationPositions = {
+			start: 0,
+			finish: websitesCount,
+		};
+
+		const itemAmountIsNumber = Number(itemsPerPage);
+
+		if (itemAmountIsNumber) {
+			paginationPositions.start = (pageNumber - 1) * itemAmountIsNumber;
+			paginationPositions.finish =
+				paginationPositions.start + itemAmountIsNumber <= websitesCount
+					? paginationPositions.start + itemAmountIsNumber
+					: websitesCount;
+		}
+
+		return paginationPositions;
+	};
+
+	getPaginationText = (
```

Comment:
I think this is a clear symptom of this being a large component - we've abstracted things out which makes sense to make the render method smaller. But as a result, this is something that's very close to JSX but not quite which doesn't quite fit with being React.